app/controllers/project.py

```diff
@@ -863,3 +872,98 @@ def post(self, project_id):
             status='success',
             message=f'project {project_id} enabled successfully'
         ), 201
+
+
+class ProjectPinView(Resource):
+    @jwt_required
+    def post(self , project_id):
+
+        current_user_id = get_jwt_identity()
+
+        project = Project.get_by_id(project_id)
+        user = User.get_by_id(current_user_id)
+
+        if (not user):
+            return dict(
+                status = 'fail',
+                message = 'The user does not exist'
+            ),404
+        
+        if (project not in user.projects):
+            return dict(
+                message = 'Project is not this users project',
+                status = 'fail'
+            ),409
+        
+        if (not project):
+            return dict(
+                status = 'fail',
+                message = 'The project id provided does not exist'
+            ),404
+        
+        if (project.pinned):
+            return dict(
+                status = 'fail',
+                message = 'The project was already pinned'
+            ),409
+        
+        
+
+        pinned_projects = [project for project in user.projects if (project.pinned != 'null' or project.pinned )]
```

Comment:
Getting an entire users list of projects and looping through it might be costly. Rather do a query against the database
```suggestion
        pinned_projects_count = Project.count(Project.owner_id = current_user_id, pinned=True)
```